processor/spanmetricsprocessor/config.go

```diff
@@ -20,14 +20,14 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-// Dimension defines the dimension name and optional default value if the Dimension is missing from a span attribute.
-type Dimension struct {
+// dimension defines the dimension name and optional default value if the dimension is missing from a span attribute.
+type dimension struct {
 	Name    string  `mapstructure:"name"`
 	Default *string `mapstructure:"default"`
 }
 
-// Config defines the configuration options for spanmetricsprocessor.
-type Config struct {
+// spanMetricsConfig defines the configuration options for spanmetricsprocessor.
+type spanMetricsConfig struct {
```

Comment:
I wasn't sure if we'd want to export `Config` for godoc generation as I've already documented configuration in the README. But if the convention is to document configs through godoc, happy to just close this PR and leave things as is.

Reply:
For reference, here is the current godoc for spanmetricsprocessor: https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanmetricsprocessor

Reply:
I do not know the answer to this. @bogdandrutu?

Reply:
I also don't know about this. We should probably file an issue to discuss with the community if this is needed.

Reply:
Yeah, when creating this processor, I was following the `routingprocessor`'s lead in exporting the `Config`.

Reply:
I ended up having Config public for other components, which is what makes the most sense to me.

Reply:
Thanks @jpkrohling, in that case I'll close this PR and corresponding issue.